qfieldsync/core/cloud_api.py

```diff
@@ -682,14 +682,16 @@ def get_last_login_error(self) -> str:
 
         error = self.user_details.get("error")
         if self._login_error:
-            if self._login_error.httpCode is None:
+            if self._login_error.httpCode is None or (
+                self._login_error.httpCode >= 400 and self._login_error.httpCode < 500
+            ):
                 error = str(self._login_error)
             elif self._login_error.httpCode >= 500:
                 error = self.tr("Server error {}").format(self._login_error.httpCode)
```

Comment:
```suggestion
            http_code = self._login_error.httpCode
            if http_code and http_code >= 500:
                error = self.tr("Server error {}").format(http_code)
            elif http_code is None or http_code >= 400:
                error = str(self._login_error)
```

Reply:
just some idea, not sure bout correctness But creating a variable would improve readability anyway?

Reply:
Agreed. I am not a big fan of reassigning values to new variables, but in this case it is better. Thanks!